index.js

```diff
@@ -57,10 +58,16 @@ function cookieSession (options) {
 
   debug('session options %j', opts)
 
+  var algorithm = opts.algorithm || 'sha1'
+  var encoding = opts.encoding || 'base64'
+
+  var cookieOptions
+  if (keys) {
+    cookieOptions = new Keygrip(keys, algorithm, encoding)
```

Comment:
So you used to be able to provide an already-constructed `Keygrip` instance as the `keys` option. This line is breaking that because it's passing the `keys` argument into the constructor even if it's already a `Keygrip` instance.